src/citrine/seeding/find_or_create.py

```diff
@@ -91,3 +93,32 @@ def default_provider():
             return dataset_collection.register(Dataset(dataset_name, "seed summ.", "seed desc."))
         dataset = get_by_name_or_create(dataset_collection, dataset_name, default_provider)
     return dataset
+
+
+def create_or_update(collection: Collection,
+                     resource: CreationType)->CreationType:
+    """Update a resource of a given name belonging to a collection.
+    Create it if it doesn't exist. 
+    Parameters
+    ----------
+    collection : Collection
+        Collection within which you want to update or create a resource
+    resource : CreationType
+        Resource that you want to create or update. If collection already contains
+        a resource with the same name it will be updated. Otherwise it will be created.
+    Returns
+    -------
+    ResourceType
+        Registered updated or created resource.
+    """    
+    try:
+        old_resource = get_by_name_or_raise_error(collection, resource.name)
+        print(f"Updating module: {resource.name}")
+        # Copy so that passed-in resource is unaffected
+        new_resource = deepcopy(resource)
+        new_resource.uid = old_resource.uid
+        return collection.update(new_resource)
+    except ValueError:
```

Comment:
We should narrow this exception: subclass `ValueError`, throw that subclass in `get_by_name_or_raise_error`, and catch only that exception here.

Reply:
An alternative would be to call the simpler [find collection](https://github.com/CitrineInformatics/citrine-python/blob/master/src/citrine/seeding/find_or_create.py#L6) method and create when it returns `None`.  That's probably better than exception handling (since this isn't an "exceptional" case).

Reply:
Good suggestion with the find_collection. We've adjusted things as you recommended.